internal/constants/scripts.go

```diff
@@ -196,3 +196,22 @@ if [ "$host_dir" ]
 fi
```

Comment:
The `rm -f ${ETC_NETWORK_MANAGER}/*` is probably wrong now (reading the comments).

Though it shouldn't be needed in the flow where we are using `coreos-copy-firstboot-network.service` See [the code](https://github.com/coreos/fedora-coreos-config/blob/773528175403f8b6f78c70f74f0480bc6b00c038/overlay.d/05core/usr/lib/dracut/modules.d/35coreos-network/coreos-copy-firstboot-network.sh#L16-L18).

Reply:
@dustymabe I am reluctant to change the behavior of the script because it is used also with the full ISO (where we don't use `coreos-copy-firstboot-network.service`) and with older versions of OpenShift (4.12, with RHEL 8, for example). But I can do something like this:

```diff
diff --git a/internal/constants/scripts.go b/internal/constants/scripts.go
index 7182fedf5..e2f7179fb 100644
--- a/internal/constants/scripts.go
+++ b/internal/constants/scripts.go
@@ -53,8 +53,10 @@ declare -A logical_nic_mac_map
 # Find destination directory based on ISO mode
 if [[ -f /etc/initrd-release ]]; then
   ETC_NETWORK_MANAGER="/etc/coreos-firstboot-network"
+  CLEAN_ETC_NETWORK_MANAGER="true"
 else
   ETC_NETWORK_MANAGER="/etc/NetworkManager/system-connections"
+  CLEAN_ETC_NETWORK_MANAGER="false"
 fi
 echo "Info: ETC_NETWORK_MANAGER was set to $ETC_NETWORK_MANAGER"
 
@@ -188,7 +190,10 @@ if [ "$host_dir" ]
   then
     # remove default connection file create by NM(nm-initrd-generator). This is a WA until
     # NM is back to supporting priority between nmconnections
-    rm -f ${ETC_NETWORK_MANAGER}/*
+    if [ "${CLEAN_ETC_NETWORK_MANAGER}" == "true" ]
+    then
+      rm -f ${ETC_NETWORK_MANAGER}/*
+    fi
     echo "Removing default connection files in '$ETC_NETWORK_MANAGER'"
     set_logical_nic_mac_mapping
     update_interface_names_by_mapping_file
```

Would that address your concern?

Reply:
If `coreos-copy-firstboot-network` is also removing these files, does it actually hurt to leave this here or is it just that it's not necessary?

Reply:
> If `coreos-copy-firstboot-network` is also removing these files, does it actually hurt to leave this here or is it just that it's not necessary?

It doesn't hurt to leave it here, but can be quite confusing to others and your future self as well :).
You can leave the code the same, but I would highly recommend a comment to explain things a bit.

Reply:
Added some comments explaining that, please check again.